components/playlist/browser/resources/components/contextualMenu.tsx

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import styled from 'styled-components'
+
+import Icon from '@brave/leo/react/icon'
+import { color } from '@brave/leo/tokens/css'
+
+interface MenuItemProps {
+  name: string
+  iconName: string
+  onClick: () => void
+}
+
+interface MenuProps {
+  items: MenuItemProps[]
+  onShowMenu?: () => void
+  onDismissMenu?: () => void
+}
+
+const StyledRow = styled.button`
+  display: contents;
+  text-align: start;
+  white-space: nowrap;
+  cursor: pointer;
+  /* TODO(sko) Check if we need hovered background */
+`
+
+const StyledMenuIcon = styled(Icon)`
+  --leo-icon-size: 16px;
+`
+
+function ContextualMenuItem ({
+  name,
+  iconName,
+  onClick,
+  onDismiss
+}: MenuItemProps & { onDismiss: () => void }) {
+  return (
+    <StyledRow
+      onClick={() => {
+        onClick()
+        onDismiss()
+      }}
+    >
+      <span>{name}</span>
+      <StyledMenuIcon name={iconName} />
+    </StyledRow>
+  )
+}
+
+const StyledContextualMenu = styled.div`
+  position: absolute;
+  right: 0;
+
+  display: grid;
+  grid-template-columns: 1fr 16px;
+  grid-auto-rows: 28px;
+  width: fit-content;
+  gap: 0px 8px;
+  align-items: center;
+
+  padding: 8px 16px;
+  border-radius: 8px;
+  border: 1px solid var(--leo-color-gray-20);
+  background: ${color.white};
+`
+
+function ContextualMenu ({
+  items,
+  onDismiss
+}: MenuProps & { onDismiss: () => void }) {
```

Comment:
`MenuProps` has an optional `onDismissMenu` prop - is that different to this one?

Reply:
Ah, yeah, `onDismiss` here is more like 'onClickOutside' for weaving ContextualMenu and AnchorButton. I didn't want to expose this to clients, so separated from the MenuProps. same [for sko/pl-menu](https://github.com/brave/brave-core/pull/19482/files#r1280104526)

I feel like it'd be better to rename this to `onClickOutside`